packages/babel-traverse/src/path/index.ts

```diff
@@ -271,10 +279,28 @@ type NodePathMixins = typeof NodePath_ancestry &
   typeof NodePath_family &
   typeof NodePath_comments;
 
+// @ts-ignore TS throws because ensureBlock returns the body node path
+// however, we don't use the return value and treat it as a transform and
+// assertion utilities. For better type inference we annotate it as an
+// assertion method
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 interface NodePath<T>
   extends NodePathAssetions,
     NodePathValidators,
-    NodePathMixins {}
+    NodePathMixins {
+  /**
+   * @see ./conversion.ts for implementation
+   */
+  ensureBlock<
+    T extends
+      | t.Loop
+      | t.WithStatement
+      | t.Function
+      | t.LabeledStatement
+      | t.CatchClause,
+  >(
+    this: NodePath<T>,
+  ): asserts this is NodePath<T & { body: t.BlockStatement }>;
```

Comment:
`ensureBlock` now asserts the shape `{ body: t.BlockStatement }`, however TS requires explicit type annotations for assertions so I added some type annotations in other plugins.

Reply:
We could also change it to return undefined in Babel 8 maybe.